### PR TITLE
[DF] Throw if the same column is passed multiple times to Snapshot

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

File: tree/dataframe/src/RDFInterfaceUtils.cxx [hunk 1]
```diff
@@ -847,6 +847,19 @@ std::vector<bool> FindUndefinedDSColumns(const ColumnNames_t &requestedCols, con
    return mustBeDefined;
 }
 
+void CheckForDuplicateSnapshotColumns(const ColumnNames_t &cols)
+{
+   std::set<std::string> uniqueCols;
```
Comment (Axel-Naumann) [inline tree/dataframe/src/RDFInterfaceUtils.cxx:852]:
```suggestion
   std::unordered_set<std::string> uniqueCols;
```
?

**Review by Axel-Naumann — APPROVED** [review]
:heart:

File: tree/dataframe/src/RDFInterfaceUtils.cxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/src/RDFInterfaceUtils.cxx:852]: okaaaay, for the >20 columns cases https://quick-bench.com/q/GVs0W7VRxV1xSp48zOsiFZNcCJI

Comment (eguiraud) [reply tree/dataframe/src/RDFInterfaceUtils.cxx]: Done

**Pull request merged**